File: app/src/main/java/dev/ragnarok/fenrir/mvp/view/ILocalPhotosView.java
package dev.ragnarok.fenrir.mvp.view;

import androidx.annotation.NonNull;

import java.util.ArrayList;
import java.util.List;

import dev.ragnarok.fenrir.model.LocalPhoto;
import dev.ragnarok.fenrir.mvp.core.IMvpView;


public interface ILocalPhotosView extends IMvpView, IErrorView {
    void displayData(@NonNull List<LocalPhoto> data);

    void setEmptyTextVisible(boolean visible);

    void displayProgress(boolean loading);

    void returnResultToParent(ArrayList<LocalPhoto> photos);

    void updateSelectionAndIndexes();

    void setFabVisible(boolean visible, boolean anim);

    void requestReadExternalStoragePermission();
}

File: app/src/main/kotlin/dev/ragnarok/fenrir/mvp/view/ILocalPhotosView.kt
package dev.ragnarok.fenrir.mvp.view

import dev.ragnarok.fenrir.model.LocalPhoto
import dev.ragnarok.fenrir.mvp.core.IMvpView

interface ILocalPhotosView : IMvpView, IErrorView {
    fun displayData(data: List<LocalPhoto>)
    fun setEmptyTextVisible(visible: Boolean)
    fun displayProgress(loading: Boolean)
    fun returnResultToParent(photos: ArrayList<LocalPhoto>)
    fun updateSelectionAndIndexes()
    fun setFabVisible(visible: Boolean, anim: Boolean)
    fun requestReadExternalStoragePermission()
}